### PR TITLE
Upgrading to rs 0.41.3

### PR DESCRIPTION
**Pull request by Bidek56**
Upgrading to rs-0.41.3

### CODE REVIEW

**Comment by Bidek56** [comment]
@universalmind303 
The only missing test from this PR is `expr.test.ts:expr.slice`
If you have any ideas how to fix `polars/lazy/expr/slice` pls LMK thx.

File: package.json [hunk 1]
```diff
@@ -54,18 +54,18 @@
     "precommit": "yarn lint && yarn test"
   },
   "devDependencies": {
-    "@biomejs/biome": "=1.7.3",
-    "@napi-rs/cli": "^2.18.3",
+    "@biomejs/biome": "=1.8.3",
```
Comment (universalmind303) [inline package.json:57]: we had. some issues last time we bumped biome. I'd rather have this done as it's own PR.

File: polars/lazy/expr/index.ts [hunk 2]
```diff
@@ -887,6 +887,17 @@ export interface Expr
...
+    old: Expr | string | number | (number | string)[],
+    new_: Expr | string | number | (number | string)[],
+  ): Expr;
+  replace({
+    old,
+    new_,
+  }: {
+    old: unknown | Expr | string | number | (number | string)[];
+    new_?: Expr | string | number | (number | string)[];
+  }): Expr;
```
Comment (universalmind303) [inline polars/lazy/expr/index.ts:900]: add docstrings

File: polars/lazy/expr/index.ts [hunk 3]
```diff
@@ -887,6 +887,17 @@ export interface Expr
     default_?: Expr | string | number | (number | string)[];
     returnDtype?: DataType;
   }): Expr;
+  replace(
+    old: Expr | string | number | (number | string)[],
+    new_: Expr | string | number | (number | string)[],
```
Comment (universalmind303) [inline polars/lazy/expr/index.ts:892]:
```suggestion
    oldValues: Expr | string | number | (number | string)[],
    newValues: Expr | string | number | (number | string)[],
```

File: package.json (hunk 1 above)
Comment (Bidek56) [reply package.json:57]: The only issue I had was this option: `--apply-unsafe` has changed to `--write --unsafe`

File: polars/lazy/expr/index.ts (hunk 3 above)
Comment (Bidek56) [reply polars/lazy/expr/index.ts:892]: This change would involve changing existing options. 
We would need to deprecate `new_` and 'old' options and they would be different from Python Polars.

Comment (universalmind303) [reply polars/lazy/expr/index.ts:952]: that should be fine. It's fine if we deviate from python where it makes sense. `new` is a reserved word in JS, so i'd like to avoid it, and think `newValues` is better than `new_`.

File: polars/lazy/expr/index.ts [hunk 4]
```diff
@@ -1565,7 +1636,21 @@ export const _Expr = (_expr: any): Expr => {
...
       return _Expr(_expr.repeatBy(e));
     },
-    replace(old, newValue, defaultValue, returnDtype) {
+    replace(old, newValue) {
+      let oldIn: any = old;
+      let newIn = newValue;
+      if (old && typeof old === "object" && !Array.isArray(old)) {
+        oldIn = Object.keys(old["old"]);
+        newIn = Object.values(old["old"]);
+      }
```
Comment (universalmind303) [inline polars/lazy/expr/index.ts:1645]: FWIW, this is what I like to do for the object options

```ts
replace(oldValues, newValues) {
  if (oldValues && oldValues?.oldValues !== undefined) {
    return this.replace(old.old, old.newValues
  }
  // continue with the rest of the function
}
```

**Comment by Bidek56** [comment]
Is this conversion from number to either float or int too dangerous of an assumption? It works, but....
```
ValueType::Number => { 
    let napi_float = f64::from_napi_value(env, napi_val)?;
    if napi_float.fract() != 0.0 {
        AnyValue::Float64(napi_float)
    } else {
        AnyValue::Int32(i32::from_napi_value(env, napi_val)?)                    
    }
},
```

**Comment by universalmind303** [comment]
> Any ideas how to convert this [line](https://github.com/pola-rs/nodejs-polars/blob/5e20abc4e6f1f6c6f5d80ad3db954153d43e8247/src/datatypes.rs#L165): `ValueType::Number => AnyValue::Float64(f64::from_napi_value(env, napi_val)?),` to return `u32` or `f64` ? Thx Apparently it's not an easy task according to [SO](https://stackoverflow.com/questions/58586425/how-to-check-a-napi-value-of-type-napi-number-is-an-integer-or-decimal-by-using).

why can't it just continue to always return f64?

**Comment by Bidek56** [comment]
> > Any ideas how to convert this [line](https://github.com/pola-rs/nodejs-polars/blob/5e20abc4e6f1f6c6f5d80ad3db954153d43e8247/src/datatypes.rs#L165): `ValueType::Number => AnyValue::Float64(f64::from_napi_value(env, napi_val)?),` to return `u32` or `f64` ? Thx Apparently it's not an easy task according to [SO](https://stackoverflow.com/questions/58586425/how-to-check-a-napi-value-of-type-napi-number-is-an-integer-or-decimal-by-using).
> 
> why can't it just continue to always return f64?

Because rs-0.41.3 has introduced a int check for the slice method  parameters. Passing f64 causes an error.

**Comment by universalmind303** [comment]
@Bidek56 can't we just cast from f64 to an accepted type

```rs
#[napi(catch_unwind)]
    pub fn slice(&self, offset: &JsExpr, length: &JsExpr) -> JsExpr {
        self.inner
            .clone()
            .slice(offset.inner.clone().cast(DataType::Int64), length.inner.clone().cast(DataType::Int64))
            .into()
    }
```

**Review by universalmind303 — APPROVED** [review]
